components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodStartupActivity.kt

```diff
@@ -0,0 +1,129 @@
+import com.intellij.psi.PsiManager
+import com.intellij.util.application
+import io.gitpod.jetbrains.remote.utils.GitpodConfig
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.YamlKey.jetbrains
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.YamlKey.vmOptions
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.defaultXmxOption
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.gitpodYamlFile
+import io.gitpod.jetbrains.remote.utils.GitpodConfig.gitpodYamlReferenceLink
+import org.jetbrains.yaml.YAMLFileType
+import org.jetbrains.yaml.YAMLUtil
+import org.jetbrains.yaml.psi.YAMLFile
+import org.jetbrains.yaml.psi.YAMLKeyValue
+import java.nio.file.Paths
+
+class GitpodStartupActivity : StartupActivity.Background {
+
+    override fun runActivity(project: Project) {
+        application.invokeLater { checkMemoryOption(project) }
+    }
+
+    private fun checkMemoryOption(project: Project) {
+        val productCode = BuildNumber.currentVersion().productCode
+        val productName = GitpodConfig.getJetBrainsProductName(productCode) ?: return
+        // it's ok for .gitpod.yml to not exist
+        var vmOptions = emptyList<String>()
+        getGitpodYamlVirtualFile(project)?.let {
+            val vmOptionsKeyValue = getGitpodYamlVMOptionsPsiElement(project, it, productName)
+            vmOptions = vmOptionsKeyValue?.valueText?.split("\\s".toRegex()) ?: emptyList()
+        }
+        // if there is no -Xmx option from .gitpod.yml, compare runtime maxHeapSize with default value
+        var xmxVmOptions = vmOptions.filter { it.startsWith("-Xmx") }
+        if (xmxVmOptions.isEmpty()) {
+            xmxVmOptions = listOf(defaultXmxOption)
+        }
+        // the rightmost -Xmx option is the one to take effect (after deduplication)
+        val finalXmx = xmxVmOptions.last()
+        // shift right 20 (xmxInBytes >> 20) to convert to MiB
+        val finalXmxValueInMiB = VMOptions.parseMemoryOption(finalXmx.substringAfter("-Xmx")).shr(20)
+        val runtimeXmxValueInMiB = Runtime.getRuntime().maxMemory().shr(20)
+        if (finalXmxValueInMiB != runtimeXmxValueInMiB) {
+            showEditVMOptionsNotification(project, runtimeXmxValueInMiB, finalXmxValueInMiB, productName)
+        }
+    }
+
+    private fun getGitpodYamlVirtualFile(project: Project): VirtualFile? {
+        val basePath = project.basePath ?: return null
+        return VfsUtil.findFile(Paths.get(basePath, gitpodYamlFile), true)
+    }
+
+    private fun getGitpodYamlVMOptionsPsiElement(
+        project: Project,
+        virtualFile: VirtualFile,
+        productName: String
+    ): YAMLKeyValue? {
+        val psiFile = PsiManager.getInstance(project).findFile(virtualFile) as? YAMLFile ?: return null
+        return YAMLUtil.getQualifiedKeyInFile(psiFile, jetbrains, productName, vmOptions)
+    }
+
+    private fun showEditVMOptionsNotification(project: Project, runtimeXmxMiB: Long, configXmxMiB: Long, productName: String) {
```

Comment:
I don't think we need notificaiotns. They are too loud. Can we use inspections instead? https://plugins.jetbrains.com/docs/intellij/code-inspections.html